src/main/java/edu/kis/powp/jobs2d/command/CommandBoundariesCheckVisitor.java

```diff
@@ -0,0 +1,43 @@
+package edu.kis.powp.jobs2d.command;
+
+import java.util.Iterator;
+
+import static java.lang.Math.abs;
+
+public class CommandBoundariesCheckVisitor implements Visitor{
+	private final ICanvas canvas;
+	private boolean isExceedingBoundaries = false;
+
+	public CommandBoundariesCheckVisitor(ICanvas canvas) {
+		this.canvas = canvas;
+	}
+
+	@Override
+	public void visitICompoundCommand(ICompoundCommand iCompoundCommand) {
+		Iterator<DriverCommand> iterator = iCompoundCommand.iterator();
+
+		while(iterator.hasNext()) {
+			DriverCommand driverCommand = iterator.next();
+
+			driverCommand.accept(this);
```

Comment:
notice that you can short-cut (break) the visitation if isExceedingBoundaries is set to true.